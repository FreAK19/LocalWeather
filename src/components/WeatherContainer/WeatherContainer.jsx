import React from 'react';
import PropTypes from 'prop-types';
import Temperature from '../Temperature/Temperature';
import WeatherStatus from '../WeatherStatus/WeatherStatus';
import Details from '../Details/Details';
import DayTime from '../DayTime/DayTime';
import getIcon from '../../utils/getIcon';
import './WeatherContainer.less';

const WeatherContainer = ({ weather, error }) => (
  <div className="weather">
    <div className="weather__header flex">
      <DayTime caption={weather.name} error={error} />
      <WeatherStatus icon={getIcon(weather.weather[0].id)} status={weather.weather[0].description} />
    </div>
    <div className="weather__footer flex">
      <Temperature temp={weather.main.temp} />
      <Details details={Object.assign({}, weather.main, weather.wind)} />
    </div>
  </div>
);

WeatherContainer.propTypes = {
  weather: PropTypes.object.isRequired,
  error: PropTypes.any
};

export default WeatherContainer;
