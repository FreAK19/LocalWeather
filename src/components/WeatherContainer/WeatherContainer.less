@bg-color: rgb(49, 62, 69);
@bg-color-viewport: rgb(50, 50, 50);
@padding-weather-header: 3.5rem 3.121rem 1.55rem 4.735rem;
@shadow: 0 2px 10px rgba(0, 0, 0, .3);
@viewport-size-tablet: 640px;
@viewport-size-phone: 510px;

.weather {
  margin: 0 auto;
  background: @bg-color;
  color: #fff;
  width: 600px;
  text-align: center;
  border-radius: 25px;
  box-shadow: @shadow;

  &__header {
    padding: @padding-weather-header;
  }

  &__footer {
    border-top: 4px solid rgb(38, 50, 56);
  }

  @media all and (max-width: @viewport-size-tablet) {
    & {
      background: @bg-color-viewport;
      border-radius: 0;
      width: 80vw;
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: 2rem 0;
    }

    &__header {
      padding: 0;
      flex-basis: 50%;
    }

    &__footer {
      border-top: transparent;
      flex-basis: 50%;
    }
  }

  @media all and (min-width: @viewport-size-tablet) {
    & {
      min-width: 460px;
    }
  }

  @media screen and (max-width: @viewport-size-phone) {
    & {
      flex-direction: column;
      padding: 4rem 0;
    }
  }
}

.flex {
  display: flex;
  justify-content: space-between;

  @media screen and (max-width: @viewport-size-phone) {
    & {
      display: block;
    }
  }
}