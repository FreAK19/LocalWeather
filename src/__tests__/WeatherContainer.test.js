import React from 'react';
import WeatherContainer from '../components/WeatherContainer/WeatherContainer'

test('WeatherContainer should render correctly', () => {
  const data = {
    main: {
      temp: 0,
        pressure: 0,
        humidity: 0,
        temp_min: 0,
        temp_max: 0
    },
    name: '',
      wind: {
      speed: 0
    },
    weather: [
      {
        id: 800,
        main: 'Clear',
        description: 'Weather Status'
      }
    ]
  };
  const component = shallow(<WeatherContainer error={false} weather={data}/>);
  expect(component).toMatchSnapshot();
});

test('WeatherContainer has two children elements', () => {
  const data = {
    main: {
      temp: 0,
      pressure: 0,
      humidity: 0,
      temp_min: 0,
      temp_max: 0
    },
    name: '',
    wind: {
      speed: 0
    },
    weather: [
      {
        id: 800,
        main: 'Clear',
        description: 'Weather Status'
      }
    ]
  };
  const component = shallow(<WeatherContainer error={false} weather={data}/>);
  expect(component.children().length).toBe(2);
});
