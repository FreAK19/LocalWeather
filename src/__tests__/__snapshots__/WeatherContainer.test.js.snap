// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherContainer should render correctly 1`] = `
<div
  className="weather"
>
  <div
    className="weather__header flex"
  >
    <DayTime
      caption=""
      error={false}
    />
    <WeatherStatus
      icon="pe-is-w-moon-2"
      status="Weather Status"
    />
  </div>
  <div
    className="weather__footer flex"
  >
    <Temperature
      temp={0}
    />
    <Details
      details={
        Object {
          "humidity": 0,
          "pressure": 0,
          "speed": 0,
          "temp": 0,
          "temp_max": 0,
          "temp_min": 0,
        }
      }
    />
  </div>
</div>
`;
